### PR TITLE
Fix 784

### PR DESCRIPTION
**Pull request by cpitclaudel**
Edit: Connects to #784

### CODE REVIEW

File: flycheck.el [hunk 1]
```diff
@@ -1803,14 +1802,15 @@ CHECKER is a checker symbol.
 
 Pop up a help buffer with the documentation of CHECKER."
   (interactive
-   (let* ((checker (flycheck-checker-at-point))
+   (let* ((checker-at-pt (flycheck-checker-at-point))
           (enable-recursive-minibuffers t)
```
Comment (lunaryorn) [inline flycheck.el]: I wonder what I added this binding for?  Do we actually use recursive mini buffers here?!

File: flycheck.el [hunk 2]
```diff
@@ -1803,14 +1802,15 @@ CHECKER is a checker symbol.
 
 Pop up a help buffer with the documentation of CHECKER."
   (interactive
-   (let* ((checker (flycheck-checker-at-point))
+   (let* ((checker-at-pt (flycheck-checker-at-point))
```
Comment (lunaryorn) [inline flycheck.el]: I think this binding is redundant.  It's only used at one place, so we can just as well inline it there, can't we?

File: flycheck.el [hunk 3]
```diff
@@ -1803,14 +1802,15 @@ CHECKER is a checker symbol.
...
-     (list (or reply checker))))
-  (if (or (null checker) (not (flycheck-valid-checker-p checker)))
+          (default (or checker-at-pt
+                       (ignore-errors (flycheck-get-checker-for-buffer))))
+          (prompt (if default
+                      (format "Describe syntax checker (default %s): " default)
+                    "Describe syntax checker: ")))
+     (list (read-flycheck-checker prompt default))))
+  (if (not (flycheck-valid-checker-p checker))
       (message "You didn't specify a Flycheck syntax checker.")
```
Comment (lunaryorn) [inline flycheck.el]: While we are at it, shouldn't this be a `user-error`?  I think that's the better choice here, isn't it?

**Comment by lunaryorn** [comment]
@cpitclaudel Great implementation, just a couple of minor nit-picks from myself :blush:  And, geez, I wonder what had come to my mind at some of this code… :flushed: :wink:

File: flycheck.el (hunk 1 above)
Comment (cpitclaudel) [inline flycheck.el]: Not sure; I don't really use recursive minibuffers myself :/

File: flycheck.el (hunk 2 above)
Comment (cpitclaudel) [inline flycheck.el]: Done!

File: flycheck.el (hunk 3 above)
Comment (cpitclaudel) [inline flycheck.el]: Yup. Updated it.

**Comment by lunaryorn** [comment]
@cpitclaudel I've cherry-picked your changes into `master`.  Thanks :+1:

**Pull request closed without merging**